Tribler/community/tunnel/remotes/sync_dict.py

```diff
@@ -0,0 +1,93 @@
+import logging
+
+from twisted.internet.task import LoopingCall
+
+from Tribler.community.tunnel.remotes.remote_object import RemoteObject
+
+
+class SyncDict(dict):
+
+    """A dictionary which syncs its items with other SyncDicts
+    """
+
+    def __init__(self, cls, task_manager=None, callback=lambda x: None,
+                 init_callback=None, sync_interval=5.0):
+        """Create a new sync dict for a certain class type
+
+        :param cls: the type of RemoteObject to store
+        :type cls: type
+        :param task_manager: the TaskManager to register with
+        :type task_manager: Tribler.dispersy.taskmanager.TaskManager
+        :param callback: the callback to call with serialized data
+        :type callback: func
+        :param sync_interval: how often to check for dirty objects
+        :type sync_interval: float
+        :returns: None
+        """
+        super(SyncDict, self).__init__()
+
+        assert issubclass(cls, RemoteObject)
+
+        self.cls = cls
+        self.sync_interval = sync_interval
+        self.callback = callback
+        self.init_callback = init_callback
+
+        if task_manager and sync_interval > 0:
+            self.register_task(task_manager)
+
+    def register_task(self, task_manager):
```

Comment:
I opted to attach this to the community class: a SyncDict is a field of a Community, so all events should be attached to and destroyed with the Community it belongs to.